lib/rules/no-tracked-properties-from-args.js

```diff
@@ -12,9 +14,10 @@ function visitClassPropertyOrPropertyDefinition(node, context, trackedImportName
     isClassPropertyOrPropertyDefinitionWithDecorator(node, trackedImportName);
 
   const hasThisArgsValue =
-    node.value.object?.type === 'MemberExpression' &&
-    node.value.object?.object.type === 'ThisExpression' &&
-    node.value.object?.property.name === 'args';
+    node.value &&
+    startsWithThisExpression(node.value) &&
+    nodeToDependentKey(node.value, context) &&
+    nodeToDependentKey(node.value, context).split('.')[0] === 'args';
```

Comment:
Removing unnecessary check:


```suggestion
    node.value &&
    startsWithThisExpression(node.value) &&
    nodeToDependentKey(node.value, context).split('.')[0] === 'args';
```